integration_test_util/types/temporary.go

```diff
@@ -0,0 +1,43 @@
+package types
+
+import (
+	"fmt"
+	tmtypes "github.com/tendermint/tendermint/types"
+	"path"
+	"strings"
+)
+
+type TemporaryHolder struct {
+	files                []string
+	tendermintGenesisDoc *tmtypes.GenesisDoc
+}
+
+func NewTemporaryHolder() *TemporaryHolder {
+	return &TemporaryHolder{}
+}
+
+func (h *TemporaryHolder) AddTempFile(file string) {
+	if len(file) < 1 {
+		return
+	}
+	if !strings.HasPrefix(file, "/tmp/") {
+		panic(fmt.Sprintf("temp file must be in '/tmp': %s", file))
+	}
```

Comment:
Won't work correctly with Windows, feel free to enhance if anyone in the team using Windows.